code/components/extra-natives-five/src/VehicleExtraNatives.cpp

```diff
@@ -363,4 +367,110 @@ static HookFunction initFunction([]()
 	{
 		*reinterpret_cast<float *>(wheelAddr + WheelXOffsetOffset) = context.GetArgument<float>(2);
 	}));
+
+	static struct : jitasm::Frontend {
+		static bool ShouldSkipRepairFunc(fwEntity* VehPointer)
+		{
+
+			for (auto it = vehicles.begin(); it != vehicles.end(); it++) {
+
+				if (VehPointer == *it) {
+					return true;
+				}
+			}
+			return false;
+		}
+		virtual void InternalMain() override
+		{
+			//Restore registers from call and push registers to the stack to be preserved
+			mov(rcx, rbx);
+			push(rax);
+			push(rdx);
+
+			sub(rsp, 0x28);
+			mov(rax, (uintptr_t)ShouldSkipRepairFunc);
+			call(rax);
+			add(rsp, 0x28);
+
+			pop(rdx);
+			cmp(al, true); //al is 1 if repair func should be skipped
+			je("skiprepair");
+			pop(rax);
+			sub(rsp, 0x28);
+			AppendInstr(jitasm::InstrID::I_CALL, 0xFF, 0, jitasm::Imm8(2), qword_ptr[rax + 0x5D0]);
+			add(rsp, 0x28);
+			ret();
+			L("skiprepair");
+			pop(rax);
+			ret();
+		}
+	} asmfunc;
+
+	static auto * repairFunc = hook::pattern("48 8B 03 45 33 C0 B2 01 48 8B CB FF 90 D0 05 00 00 48 8B 4B 20").get_first(11);
```

Comment:
use `hook::get_pattern`, also no need for it to be `static`.

also, `*` is implied by auto